content/shmem_test_all.tex

```diff
@@ -5,13 +5,14 @@
 \begin{apidefinition}
 
 \begin{C11synopsis}
-int @\FuncDecl{shmem\_test\_all}@(TYPE *ivars, size_t nelems, int cmp, TYPE cmp_value);
+int @\FuncDecl{shmem\_test\_all}@(TYPE *ivars, size_t nelems, const int *status, int cmp, TYPE cmp_value);
 \end{C11synopsis}
```

Comment:
Apologize for my late question. I am not sure why is the `status` argument is also added for `test_all` and `wait_until_all`. Could you please explain the reason ?

Reply:
Sure - this came up during our discussion about whether `status` should be an "IN" only; but admittedly, the change to add `status` to the "all" routines is a little tangential to the IN/OUT issue.  Anyway, the relevant use case occurs when a user wants to call wait/test on a constant subset of _all_ values in `ivars`.  An example application might be communicating boundary cell data across a regular mesh, where each PE owns some part of the mesh, and that part of the mesh consists of `ivars`.  Every iteration the relevant boundary cells (a subset within `ivars`) stay the same (which a good argument for IN-only `status`) and the user wants to wait on _all_ the elements specified by `status`.  This sort of application does not fit the "some" idiom with `status`, because "some" would only indicate whether _at least one_ element satisfies the condition.  In apps like this, you often can't continue until all the boundary cells are updated (like in particle-in-cell apps, to be more specific).

It also makes the interface a little bit more consistent, which is nice.

Reply:
Thanks, it makes sense to me.